File: skills/astra-toolkit/clients/java/examples/tables-list-metadata-default.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.definition.TableDescriptor;
import java.util.List;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient(System.getenv("ASTRA_DB_APPLICATION_TOKEN")).getDatabase(System.getenv("ASTRA_DB_API_ENDPOINT"));

    // List table metadata
    List<TableDescriptor> result = database.listTables();

    System.out.println(result);
  }
}

File: skills/astra-toolkit/clients/go/examples/tables-list-metadata-default.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"os"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get a database
	client := astra.NewClient()

	database := client.Database(
		os.Getenv("ASTRA_DB_API_ENDPOINT"),
		options.API().SetToken(os.Getenv("ASTRA_DB_APPLICATION_TOKEN")),
	)

	// List table metadata
	tables, err := database.ListTables(ctx)

	if err != nil {
		log.Fatal(err)
	}

	output, err := json.Marshal(tables)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(output))
}
